packages/tiptap/src/Editor.js

```diff
@@ -492,9 +492,9 @@ export default class Editor extends Emitter {
       return
     }
 
-    const newState = this.state.reconfigure({
-      plugins: this.state.plugins.concat([plugin]),
-    })
+    this.plugins.push(plugin);
+    this.state.plugins.splice(this.plugins.length, 0, plugin);
+    const newState = this.state.reconfigure({plugins: this.state.plugins});
```

Comment:
Should be the same with `[plugin, ...this.state.plugins]`, right? 🤔

Reply:
Pushing new plugins to the front of `this.state.plugins` could also work, as long as they're in-front of the default ProseMirror plugins.

An alternative to accommodate both of these solutions would be to modify parameters of `registerPlugin`, and to allow a helper function to be passed in, e.g.
```js
  registerPlugin(plugin, helper) {
    let plugins = typeof helper == 'function' ? function(plugin, this.state.plugins) : [...plugin, this.state.plugins];
    const newState = this.state.reconfigure({plugins});
    this.view.updateState(newState)
  }
```

This would allow custom behavior for where new plugins should be registered within `this.state.plugins`. An example is if plugins had to be registered in a certain order. Thoughts?

Reply:
This is great! Lets do this!

Reply:
Ok, I've updated the PR to allow a function to be passed in. :)